controllers/vspherevm_controller.go

```diff
@@ -115,13 +115,13 @@ func AddVMControllerToManager(ctx *context.ControllerManagerContext, mgr manager
 			UpdateFunc: func(e event.UpdateEvent) bool {
 				oldCluster := e.ObjectOld.(*clusterv1.Cluster)
 				newCluster := e.ObjectNew.(*clusterv1.Cluster)
-				return oldCluster.Spec.Paused && !newCluster.Spec.Paused
```

Comment:
This is a discrepancy, we should check for the `paused` field (only `newCluster` for UpdateEvent) as well as the presence of the annotation.

Reply:
Thanks!